### PR TITLE
fix(core): Template builder autosave multiple times

### LINKED ISSUE #27689: Layout Editor: auto save data is messing content
### Problem Statement
In the layout editor, when a row is moved and autosaved, then another row is moved and "Content" is clicked right away, the content becomes disorganized.

### Steps to Reproduce
1. Open the layout editor in dotCMS page.
2. Move a row and allow it to autosave.
3. Move another row and allow it to autosave.
4. Click on "Content."
5. Observe the content move to another containers.

https://github.com/dotCMS/core/assets/751424/58da8e4c-8e55-4217-ac1b-68cf828819b5

I was able to repro by just doing the requests myself, one after the other:

```shell
curl 'http://localhost:8080/api/v1/page/739885e5c62368bafb2abe97aa70fb9e/layout?variantName=DEFAULT' \
  -H 'Accept: */*' \
  -H 'Accept-Language: en-US,en;q=0.9' \
  -H 'Cache-Control: no-cache' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: JSESSIONID=F070104DAD10E0BAD3B2E62DD1503098; access_token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJqdGkiOiI3MTgxNjdjOC1jN2ZmLTQyYzAtOWY1MC1mMjdlMjI4NWU2ZTMiLCJ4bW9kIjoxNzA4NjA1OTc2OTk4LCJzdWIiOiJkb3RjbXMub3JnLjEiLCJpYXQiOjE3MDg2MDU5NzcsImlzcyI6IjhjZTU1ZmI1NWUiLCJleHAiOjE3MDg2OTIzNzd9.43xnumRXFNbd5WT1yVPWLawNnjYbgT6jL22qxPk0SM4; DWRSESSIONID=bAVj1dcONYoHCx45scOMtdHSgTo' \
  -H 'DNT: 1' \
  -H 'Origin: http://localhost:8080' \
  -H 'Pragma: no-cache' \
  -H 'Referer: http://localhost:8080/dotAdmin/' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36' \
  -H 'sec-ch-ua: "Chromium";v="121", "Not A(Brand";v="99"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  --data-raw '{"themeId":"SYSTEM_THEME","layout":{"width":"","title":"anonymouslayout1708557409935","header":true,"footer":true,"body":{"rows":[{"columns":[{"containers":[{"identifier":"SYSTEM_CONTAINER","uuid":"4"}],"leftOffset":1,"width":6,"styleClass":""},{"containers":[{"identifier":"SYSTEM_CONTAINER","uuid":"5"}],"leftOffset":7,"width":6,"styleClass":""}],"styleClass":null},{"columns":[{"containers":[{"identifier":"SYSTEM_CONTAINER","uuid":"1"}],"leftOffset":1,"width":4,"styleClass":"hello"},{"containers":[{"identifier":"SYSTEM_CONTAINER","uuid":"2"}],"leftOffset":5,"width":4,"styleClass":""},{"containers":[{"identifier":"SYSTEM_CONTAINER","uuid":"3"}],"leftOffset":9,"width":4,"styleClass":""}],"styleClass":null}]},"sidebar":null},"title":null}' \
  --compressed
```

and...

```shell
curl 'http://localhost:8080/api/v1/page/739885e5c62368bafb2abe97aa70fb9e/layout?variantName=DEFAULT' \
  -H 'Accept: */*' \
  -H 'Accept-Language: en-US,en;q=0.9' \
  -H 'Cache-Control: no-cache' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: JSESSIONID=F070104DAD10E0BAD3B2E62DD1503098; access_token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJqdGkiOiI3MTgxNjdjOC1jN2ZmLTQyYzAtOWY1MC1mMjdlMjI4NWU2ZTMiLCJ4bW9kIjoxNzA4NjA1OTc2OTk4LCJzdWIiOiJkb3RjbXMub3JnLjEiLCJpYXQiOjE3MDg2MDU5NzcsImlzcyI6IjhjZTU1ZmI1NWUiLCJleHAiOjE3MDg2OTIzNzd9.43xnumRXFNbd5WT1yVPWLawNnjYbgT6jL22qxPk0SM4; DWRSESSIONID=bAVj1dcONYoHCx45scOMtdHSgTo' \
  -H 'DNT: 1' \
  -H 'Origin: http://localhost:8080' \
  -H 'Pragma: no-cache' \
  -H 'Referer: http://localhost:8080/dotAdmin/' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36' \
  -H 'sec-ch-ua: "Chromium";v="121", "Not A(Brand";v="99"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  --data-raw '{"themeId":"SYSTEM_THEME","layout":{"width":"","title":"anonymouslayout1708557409935","header":true,"footer":true,"body":{"rows":[{"columns":[{"containers":[{"identifier":"SYSTEM_CONTAINER","uuid":"1"}],"leftOffset":1,"width":4,"styleClass":"hello"},{"containers":[{"identifier":"SYSTEM_CONTAINER","uuid":"2"}],"leftOffset":5,"width":4,"styleClass":""},{"containers":[{"identifier":"SYSTEM_CONTAINER","uuid":"3"}],"leftOffset":9,"width":4,"styleClass":""}],"styleClass":null},{"columns":[{"containers":[{"identifier":"SYSTEM_CONTAINER","uuid":"4"}],"leftOffset":1,"width":6,"styleClass":""},{"containers":[{"identifier":"SYSTEM_CONTAINER","uuid":"5"}],"leftOffset":7,"width":6,"styleClass":""}],"styleClass":null}]},"sidebar":null},"title":null}' \
  --compressed
```

### Acceptance Criteria
The content should stay consistent in its respective containers, no matter how we save the page layout.

### External Links
N/A

### Assumptions & Initiation Needs
- The issue is related to data synchronization after autosave.
- Development team needs access to the layout editor code to make changes.

### PR DESCRIPTION
**Pull request by zJaaal**
## Proposed changes
* Delete extra emitter in `dot-template-builder.component` which caused a double rendering in Templates Portlet
* Add method in the store to update the layout model without changing GridStack ID
* Include method in `ngOnChanges` to update the layout with the changes that the backend returned after save.

## Templates Portlet (UUIDs added as visual representation, this is not showed in the prod version)
https://github.com/dotCMS/core/assets/63567962/801c1779-cc12-448e-8d3b-5eddbaac8e53

## Edit EMA ( The numbers I follow with the mouse are the UUIDs and are only for visual representation)
https://github.com/dotCMS/core/assets/63567962/aed3fa0a-a3cb-4954-b2e9-1723a0527d65

## Old Edit Page
https://github.com/dotCMS/core/assets/63567962/051e84f0-9587-4f73-a89d-69e809437238

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-create-edit/dot-template-builder/dot-template-builder.component.ts [hunk 1]
```diff
@@ -67,7 +67,6 @@ export class DotTemplateBuilderComponent implements OnInit, OnDestroy {
      * @memberof DotTemplateBuilderComponent
      */
     onTemplateItemChange(item: DotTemplateItem) {
-        this.updateTemplate.emit(item);
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-create-edit/dot-template-builder/dot-template-builder.component.ts:70]: No test breaking?

File: core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-box/template-builder-box.component.html [hunk 2]
```diff
@@ -65,6 +65,10 @@
             <div class="template-builder-box__item" *ngFor="let item of items; let i = index">
                 <p data-testId="container-title">
                     {{ (containerMap && containerMap[item.identifier]?.title) || item.identifier }}
+                    - uuid: {{ item.uuid }}
+                    <!--
+                    IF YOU SEE THIS COMMENT PLEASE REMIND ME TO REMOVE THE UUID FROM THE BOX. IM
+                    USING THIS FOR POC MATTERS -->
```
Comment (fmontes) [inline core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-box/template-builder-box.component.html:71]: Remove.

Comment (KevinDavilaDotCMS) [inline core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-box/template-builder-box.component.html:71]: Remove uuid (?)

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-create-edit/dot-template-builder/dot-template-builder.component.ts (hunk 1 above)
Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-create-edit/dot-template-builder/dot-template-builder.component.ts:70]: Probably I'm running tests to check what is broken

File: core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-box/template-builder-box.component.html (hunk 2 above)
Comment (zJaaal) [reply core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-box/template-builder-box.component.html]: Done.

Comment (zJaaal) [reply core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-box/template-builder-box.component.html]: Done.

**Comment by zJaaal** [comment]
We have a bug in the old edit page, will leave in draft while we fix it.

**Review by fmontes — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

**Review by manuelrojas — APPROVED** [review]

**Pull request merged**